### PR TITLE
Server api basics

### PR DESCRIPTION
**Pull request by computmaxer**
I read through all of the docs for [Flask-RESTful](http://flask-restful.readthedocs.org/en/latest/index.html) and implemented a basic API for Server objects in the database.

Creating, getting, and deleting Servers works.  Updating them and getting them with a filter does not work yet.  The API actually creates and fetches in/from the database, it is fully functional in that way.

**Edit:** I also switched the Vagrant config to not user supervisord to run the app.  It is easier and better to use the standard dev server that Flask provides.  nginx proxy is still setup though

### CODE REVIEW

**Comment by computmaxer** [comment]
@mpdavis @qinxiaoapp @organman91

File: app/requirements.txt [hunk 1]
```diff
@@ -13,4 +13,5 @@ mongoengine==0.8.4
 paramiko==1.11.0
 pycrypto==2.6
 pymongo==2.6.3
+six==1.4.1
```
Comment (mpdavis) [inline app/requirements.txt:16]: Why are you adding six?

File: provision/run_dev.sh [hunk 2]
```diff
@@ -1,4 +1,6 @@
 #!/bin/bash
+/bin/killall python
```
Comment (mpdavis) [inline provision/run_dev.sh:2]: Is this ever going to cause use problems by killing stuff we don't want to kill?

**Comment by mpdavis** [comment]
Nothing that keeps me from seeing this merged.  +1

File: app/requirements.txt (hunk 1 above)
Comment (computmaxer) [inline app/requirements.txt:16]: flask_restful requires it.

File: provision/run_dev.sh (hunk 2 above)
Comment (computmaxer) [inline provision/run_dev.sh:2]: Nah :)

**Comment by qinxiaoapp** [comment]
+1

File: provision/run_dev.sh (hunk 2 above)
Comment (qinxiaoapp) [inline provision/run_dev.sh:2]: Sry I merged the pull request and dont know it would be closed immediately.
Hopefully it wont hurt! D:

**Comment by mpdavis** [comment]
Merging it doesn't hurt anything here, as the code was good to go anyway.

**Pull request merged**